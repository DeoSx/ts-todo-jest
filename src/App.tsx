import * as React from 'react'

import { useReducer } from 'react'
import { Action, State, ContextState } from './types/StateTypes'
import { todoReducer } from './reducer'
import NewTask from './components/NewTask'
import TasksList from './components/TaskList'

export const initialState: State = {
  newTask: '',
  tasks: []
}

export const ContextApp = React.createContext<Partial<ContextState>>({})

const App: React.FC = () => {
  const [state, changeState] = useReducer<React.Reducer<State, Action>>(
    todoReducer,
    initialState
  )

  const ContextState: ContextState = {
    state,
    changeState
  }

  return (
    <>
      <h1>ts-react-jest</h1>
      <ContextApp.Provider value={ContextState}>
        <NewTask />
        <TasksList />
      </ContextApp.Provider>
    </>
  )
}

export default App
