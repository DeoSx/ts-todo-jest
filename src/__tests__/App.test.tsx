import * as React from 'react'
import { shallow } from 'enzyme'
import { render, cleanup, fireEvent } from '@testing-library/react'
import App from '../App'

describe('<App/>', () => {
  // clean enviroment after test
  afterEach(cleanup)

  test('hasn`t got changes', () => {
    const component = shallow(<App />)
    expect(component).toMatchSnapshot()
  })

  test('should render right input value', async () => {
    const { container } = render(<App />)
    expect(container.querySelector('input').getAttribute('value')).toEqual('')

    fireEvent.change(container.querySelector('input'), {
      target: {
        value: 'test'
      }
    })
    expect(container.querySelector('input').getAttribute('value')).toEqual(
      'test'
    )

    fireEvent.click(container.querySelector('button'))
    expect(container.querySelector('input').getAttribute('value')).toEqual('')
  })
})
