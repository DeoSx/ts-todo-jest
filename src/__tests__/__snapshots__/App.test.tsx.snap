// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> hasn\`t got changes 1`] = `
<Fragment>
  <h1>
    ts-react-jest
  </h1>
  <ContextProvider
    value={
      Object {
        "changeState": [Function],
        "state": Object {
          "newTask": "",
          "tasks": Array [],
        },
      }
    }
  >
    <NewTask />
    <TasksList />
  </ContextProvider>
</Fragment>
`;
